ddpui/celeryworkers/tasks.py

```
@@ -876,6 +920,103 @@ def summarize_logs(
         llm_session.save()
 
 
+@app.task(bind=True, max_retries=3, default_retry_delay=60)
+def trigger_log_summarization_for_failed_flow(self, flow_run_id: str, flow_run: dict = None):
+    """
+    Triggers automatic summarization for failed pipeline runs.
+    Handles both regular pipelines and airbyte syncs.
+    """
+    try:
+        # Get flow run if not provided
+        if not flow_run:
+            flow_run = get_flow_run_poll(flow_run_id)
+
+        # Check if summary already exists to prevent duplicates
+        existing_summary = LlmSession.objects.filter(
+            flow_run_id=flow_run_id, session_type=LlmAssistantType.LOG_SUMMARIZATION
+        ).exists()
+
+        if existing_summary:
+            logger.info(f"Summary already exists for flow_run {flow_run_id}")
+            return
+
+        # Get failed task information
+        task_runs = get_flow_run_graphs(flow_run_id)
+
+        # Find the failed task
+        failed_task = None
+        for task in task_runs:
+            if task.get("state_type") == "FAILED" or task.get("state_name") == "DBT_TEST_FAILED":
+                failed_task = task
+                break
+
+        if not failed_task:
+            logger.warning(f"No failed task found in flow_run {flow_run_id}")
```

you don't expect this to ever happen right, you want to raise an exception? or send yourself an email or something